engine/access/relay/engine.go

```diff
@@ -0,0 +1,111 @@
+package relay
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/network"
+	"github.com/rs/zerolog"
+)
+
+type Engine struct {
+	unit     *engine.Unit   // used to manage concurrency & shutdown
+	log      zerolog.Logger // used to log relevant actions with context
+	me       module.Local
+	conduits map[network.Channel]network.Conduit // conduits for unstaked network
+}
+
+func New(
+	log zerolog.Logger,
+	channels network.ChannelList,
```

Comment:
so instead of expecting a list of channels, what if we pass the `splitter.Network` an instance of this engine. Then, whenever, a `network.Register` call happens, the network also registers the Relay engine for the same channel.

Reply:
Based on our previous discussion, I don't think the plan was to have the relay and splitter engine to be dependent on one another. 

It doesn't really take any more work / code to decouple them, so why not do it? It will make it much easier to reuse the splitter engine in case we have another need for it in the future.